### PR TITLE
GH-1823 amCharts dynamic properties

### PR DESCRIPTION
**Pull request by mounicarachakonda**
Fixes GH-1823

signed off by :mounica.rachakonda@lexisnexis.com

### CODE REVIEW

**Comment by jbrundage** [comment]
@mounicarachakonda The **yAxisMinimumValue** and **yAxisMaximumValue** only affect the chart if both values are specified. Could you modify your code to allow these parameters to work independently?

**Comment by jbrundage** [comment]
Link to rawgit example:
http://rawgit.com/mounicarachakonda/Visualization/amcharts/demos/dermatology.html?src/amchart/Bar.simple&yAxisMinimumValue=10&yAxisMaximumValue=200

**Comment by mounicarachakonda** [comment]
@jbrundage  Please review.

**Comment by jbrundage** [comment]
The chart doesn't load correctly without your min/max values. See this: http://rawgit.com/mounicarachakonda/Visualization/amcharts/demos/dermatology.html?src/amchart/Bar.simple

**Comment by mounicarachakonda** [comment]
@jbrundage I have made the changes. Please review.

File: src/amchart/Column.js [hunk 1]
```diff
@@ -34,6 +34,12 @@
     Column.prototype.publish("stackType", "regular", "set", "Stack Type",["none","regular","100%","3d"],{tags:["Basic"]});
     Column.prototype.publish("useOhlcLines", false, "boolean", "Use OHLC Lines",null,{tags:["Intermediate"]});
 
+    Column.prototype.publish("yAxisMinimumValue", null, "number", "Y axis Minimum value",null,{});
```
Comment (GordonSmith) [inline src/amchart/Column.js]: Rename to "yAxisDomainLow" to match all the XYAxis based charts

File: src/amchart/Column.js [hunk 2]
```diff
@@ -34,6 +34,12 @@
...
+    Column.prototype.publish("yAxisMaximumValue", null, "number", "Y axis Maximum value",null,{});
```
Comment (GordonSmith) [inline src/amchart/Column.js]: Rename to "yAxisDomainHigh" to match all the XYAxis based charts

File: src/amchart/Column.js [hunk 3]
```diff
@@ -34,6 +34,12 @@
...
+    Column.prototype.publish("yAxisbaseValue", null, "number", "Y axis base value",null,{});
```
Comment (GordonSmith) [inline src/amchart/Column.js]: Rename to yAxisBaseValue

File: src/amchart/Column.js [hunk 4]
```diff
@@ -34,6 +34,12 @@
...
+    Column.prototype.publish("yAxislabelFrequency", 1,"number", "Y axis label frequency",null,{});
```
Comment (GordonSmith) [inline src/amchart/Column.js]: rename to yAxisLabelFrequency

File: src/amchart/Column.js [hunk 5]
```diff
@@ -34,6 +34,12 @@
...
+    Column.prototype.publish("yAxisGridCount", 5,"number", "Y axis grid count",null,{});
```
Comment (GordonSmith) [inline src/amchart/Column.js]: Rename to yAxisTickCount for consistency with XYAxis charts

File: src/amchart/Column.js [hunk 6]
```diff
@@ -56,6 +62,27 @@
             this._chart.valueAxes[0].stackType = "none";
         }
 
+        if(this.yAxisMinimumValue() &&  this.yAxisMinimumValue()!== null){
```
Comment (GordonSmith) [inline src/amchart/Column.js]: Change to ```if (this.yAxisMinimumValue_exists())```

File: src/amchart/Column.js [hunk 7]
```diff
@@ -56,6 +62,27 @@
...
+            this._chart.valueAxes[0].minimum = this.yAxisMinimumValue();
+        }
+
+        if(this.yAxisMaximumValue() && this.yAxisMaximumValue()!==null){
```
Comment (GordonSmith) [inline src/amchart/Column.js]: Change to if (this.yAxisMaximumValue_exists())

File: src/amchart/Column.js [hunk 8]
```diff
@@ -56,6 +62,27 @@
...
+            this._chart.valueAxes[0].maximum = this.yAxisMaximumValue();
+        }
+        this._chart.valueAxes[0].strictMinMax = false;
+
+        if(this.yAxislabelFrequency()) {
```
Comment (GordonSmith) [inline src/amchart/Column.js]: Change to if (this.yAxislabelFrequency_exists())

File: src/amchart/Column.js [hunk 9]
```diff
@@ -56,6 +62,27 @@
...
+            this._chart.valueAxes[0].labelFrequency = this.yAxislabelFrequency();
+            }
+
+        if(this.yAxisbaseValue()) {
```
Comment (GordonSmith) [inline src/amchart/Column.js]: Change to if (this.yAxisbaseValue_exists())

File: src/amchart/Column.js [hunk 10]
```diff
@@ -56,6 +62,27 @@
...
+            this._chart.valueAxes[0].baseValue = this.yAxisbaseValue();
+            }
+
+        if (this.yAxisGridCount()) {
```
Comment (GordonSmith) [inline src/amchart/Column.js]: Change to if (this.yAxisGridCount_exists())

File: src/amchart/Column.js (hunk 1 above)
Comment (GordonSmith) [inline src/amchart/Column.js]: Make optional:  ```Column.prototype.publish("yAxisMinimumValue", null, "number", "Y axis Minimum value",null,{optional:true});

File: src/amchart/Column.js (hunk 2 above)
Comment (GordonSmith) [inline src/amchart/Column.js]: make optional (see above)

File: src/amchart/Column.js (hunk 3 above)
Comment (GordonSmith) [inline src/amchart/Column.js]: make optional (see above)

File: src/amchart/Column.js (hunk 4 above)
Comment (GordonSmith) [inline src/amchart/Column.js]: make optional (see above)

File: src/amchart/Column.js (hunk 5 above)
Comment (GordonSmith) [inline src/amchart/Column.js]: make optional (see above)

**Comment by mounicarachakonda** [comment]
@GordonSmith Please review.

File: src/amchart/Column.js [hunk 11]
```diff
@@ -34,6 +34,12 @@
     Column.prototype.publish("stackType", "regular", "set", "Stack Type",["none","regular","100%","3d"],{tags:["Basic"]});
     Column.prototype.publish("useOhlcLines", false, "boolean", "Use OHLC Lines",null,{tags:["Intermediate"]});
 
+    Column.prototype.publish("yAxisDomainLow", null, "number", "Y axis Minimum value",null,{optional:true});
+    Column.prototype.publish("yAxisDomainHigh", null, "number", "Y axis Maximum value",null,{optional:true});
+    Column.prototype.publish("yAxisBaseValue", null, "number", "Y axis base value",null,{optional:true});
+    Column.prototype.publish("yAxisLabelFrequency", 1,"number", "Y axis label frequency",null,{optional:true});
+    Column.prototype.publish("yAxisTickCount", 5,"number", "Y axis grid count",null,{optional:true});
```
Comment (GordonSmith) [inline src/amchart/Column.js]: Do you want to default this to 5?  I would suggest leaving it as NULL as default?

**Comment by GordonSmith** [comment]
@mounicarachakonda Just one question - I suspect setting the number of labels to 5 by default is not a good idea?

**Comment by mounicarachakonda** [comment]
@GordonSmith In amCharts documentation the default value for gridCount is given as 5.
amDocs link - https://docs.amcharts.com/javascriptcharts/ValueAxis

**Comment by GordonSmith** [comment]
@mounicarachakonda I thought the default was for autoGridCount=true?
http://docs.amcharts.com/3/javascriptcharts/AxisBase#autoGridCount

**Comment by mounicarachakonda** [comment]
@GordonSmith I have changed the value to null. Please review.

**Comment by GordonSmith** [comment]
Looks good

**Pull request merged**